### DISCUSSION TITLE
Play audio with bytes

### DISCUSSION THREAD

**Discussion by CodeEzpro** [Rapptz/discord.py#9730] [Q&A]
does someone know how to play mp3 file with "ope("file.mp","wb")" and discord.py

**Comment by ika2kki** [comment]
hi, you can use [`discord.FFmpegPCMAudio`](https://discordpy.readthedocs.io/en/stable/api.html#discord.FFmpegAudio), for example:
```py
vc = await voice_channel.connect()
vc.play(discord.FFmpegPCMAudio("filename.mp3"))
```
`ffmpeg` needs to be in your terminal PATH and ensure you've installed discord.py with voice enabled too, instructions covered here: https://discordpy.readthedocs.io/en/stable/intro.html#installing

**Comment by CodeEzpro** [comment]
Hi,  thanks for your response but i want to use bytes to play sounds without saving it.

**Reply by ika2kki** [reply]
you can pass a file-like to the source as well, for example:
```py
import io
with open("filename.mp3", mode="rb") as f:
    stream = io.BytesIO(f.read())

vc.play(discord.FFmpegPCMAudio(stream, pipe=True))
```
note the `pipe=True` keyword
